### PR TITLE
Merge release-agua

### PR DESCRIPTION
**Pull request by tobrun**
This PR merges `release-agua` back to `master`.

##### Conflict resolve strategy:
 - Core didn't show conflicts
 - iOS/macOS, worked together with @fabian-guerra on changes
 - Android, latest snapshot changes and revert mapbox-core from `release-agua`

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -294,7 +320,7 @@ jobs:
       - deploy:
           name: Publish to Maven
           command: |
-            if [ "${CIRCLE_BRANCH}" == "master" ]; then make run-android-upload-archives ; fi
+            if [ "${CIRCLE_BRANCH}" == "release-agua" ]; then make run-android-upload-archives ; fi
```
Comment (Guardiola31337) [inline circle.yml]: @tobrun shouldn't be `master`?

Comment (tobrun) [reply circle.yml]: we continue to work from the agua release branch until that 5.2.0 is finalised, this configuration is correct for that. Not having this config on master is an issue as we have been building snapshots from master and release-agua..

File: platform/ios/CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 Mapbox welcomes participation and contributions from everyone. Please read [CONTRIBUTING.md](../../CONTRIBUTING.md) to get started.
 
-## master
+## 3.7.0
```
Comment (fabian-guerra) [inline platform/ios/CHANGELOG.md]: After looking side to side the changelog I'm not sure if we should let 3.7.0 be the header. What do you think @boundsj  @1ec5?

Comment (1ec5) [reply platform/ios/CHANGELOG.md]: This happens whenever we wait too long to merge back to master. The header should say “3.7.0”, but we’ll need to manually move some of the entries up into their own “master” section.

**Comment by tobrun** [comment]
I had to redo the merge with Expressions landing between creating this PR and trying to merge it.
Pretty confident about the binding changes, parred with @fabian-guerra on it, @jfirebaugh can you double check core?

**Pull request merged**